src/main/java/org/aksw/simba/lemming/metrics/single/edgemanipulation/VertexDegrees.java

```diff
@@ -0,0 +1,175 @@
+package org.aksw.simba.lemming.metrics.single.edgemanipulation;
+
+import org.aksw.simba.lemming.ColouredGraph;
+
+import com.carrotsearch.hppc.IntIntOpenHashMap;
+
+import grph.Grph.DIRECTION;
+import it.unimi.dsi.fastutil.ints.IntIterator;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/** Class for storing in and out degree of vertices.
+ * @author Atul
+ *
+ */
+public class VertexDegrees {
+
+	// Maps for storing in-degree and out-degree for different vertices
+	private int mMapVerticesinDegree[];
+	private int mMapVerticesoutDegree[];
+
+	/**
+	 * Initializing the Maps and calling computeVerticesDegree method.
+	 * @param clonedGraph
+	 */
+	public VertexDegrees(ColouredGraph clonedGraph) {
+		// Initialize Hash Map for storing degree of vertices
+		computeVerticesDegree(clonedGraph);
+	}
+
+	/**
+	 * Computing the in degree and out degree for all vertices in graph.
+	 * @param graph
+	 */
+	private void computeVerticesDegree(ColouredGraph graph) {
+		IntSet vertices = graph.getVertices();
+		mMapVerticesinDegree = new int[vertices.size()];
+		mMapVerticesoutDegree = new int[vertices.size()];
+		int index = 0;
+		IntIterator iterator = vertices.iterator();
+		while (iterator.hasNext()) {
+			int nextInt = iterator.nextInt();
+			int inVertexDegree = graph.getGraph().getInVertexDegree(nextInt);
+			int outVertexDegree = graph.getGraph().getOutVertexDegree(nextInt);
+
+			mMapVerticesinDegree[index] = inVertexDegree;
+			mMapVerticesoutDegree[index] = outVertexDegree;
+			index++;
+		}
+	}
+
+	/**
+	 * Returns in degree for input vertex id.
+	 * @param vertexId
+	 * @return
+	 */
+	public int getVertexIndegree(int vertexId) {
+		int inDegree = mMapVerticesinDegree[vertexId];
+		return inDegree;
```

Comment:
Why is the local variable necessary? Wouldn't this be sufficient?
```java
return mMapVerticesinDegree[vertexId];
```

Reply:
Yes, I have implemented the suggested change.